### DISCUSSION TITLE
Inconsistent behaviour with `--code` flag for `train` cli

### DISCUSSION THREAD

**Discussion by uwaisiqbal** [explosion/spaCy#9910] [Help: Other Questions] [answered]
## How to reproduce the behaviour
I am trying to train a relation extraction model using spaCy and would like to be able to call the `train` method from Python. Following the documentation here (https://spacy.io/api/cli#train-function), I am trying to using the following code:

```python
    train(config_path, output_path=model_output_path / "rel",
          overrides={"paths.train": f"{train_data_path}",
                     "paths.dev": f"{dev_data_path}",
                     "code": "custom_functions.py",
                     "components.ner.source": f"{best_ner_model_path}"
                     })
```

This results in the following error:
```bash
✘ Error parsing config overrides
code   not a section value that can be overwritten
```

However, I can use the CLI to successfully run training using the `-c` or `--code` flag to point to my `custom_functions.py` script.

There seems to be an inconsistency between the expected behaviour of the `train` method and the CLI command with the `code` override.

## Info about spaCy
- **spaCy version:** 3.2.1
- **Platform:** Darwin-20.6.0-x86_64-i386-64bit
- **Python version:** 3.7.12
- **Pipelines:** en_core_web_sm (3.2.0)

**Comment by adrianeboyd** [comment, marked as answer]
Hi, `--code` isn't an override in this sense. Overrides are only for overriding a value in the config. You want to use the `code_path` option with the method `train_cli` (or manage this import in some other way in your script):

https://github.com/explosion/spaCy/blob/93e9bf681fd6fbdd3a34e2c5df6b3a7c93b9596e/spacy/cli/train.py#L18-L54

**Comment by uwaisiqbal** [comment]
Ah ok thanks for pointing that out. Is there a way to pass in the same dictionary of `overrides` into the `train_cli` method? It looks like I'll just have to make a call to `import_code` before using the `train` method and it should work hopefully.

**Reply by adrianeboyd** [reply]
No, I think you'd have to write your own function to do exactly that, but it would be just as short as `train_cli`.

**Reply by uwaisiqbal** [reply]
Great thanks for your help! Much appreciated.
